src/mlpack/methods/edge_boxes/feature_extraction_impl.hpp

```diff
@@ -0,0 +1,1052 @@
+/**
+ * @file feature_extraction_Impl.hpp
+ * @author Nilay Jain
+ *
+ * Implementation of feature extraction methods.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_ImPL_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_ImPL_HPP
+
+
+#include "feature_extraction.hpp"
+#include <mlpack/methods/pca/pca.hpp>
+namespace mlpack {
+namespace structured_tree {
+
+/**
+ * Constructor: stores all the parameters in an object
+ * of feature_parameters class.
+ */
+template<typename MatType, typename CubeType>
+StructuredForests<MatType, CubeType>::
+StructuredForests(FeatureParameters F) : params(F) {}
+
+/*
+template<typename MatType, typename CubeType>
+MatType StructuredForests<MatType, CubeType>::
```

Comment:
I guess we can remove useless codes, let git help us memorize the codes we written